scripts/packaging/extensionsfolderpreupm.ps1

```diff
@@ -0,0 +1,80 @@
+<#
+.SYNOPSIS
+    Prepares the MRTK\Extensions folder for UPM packaging.
+.DESCRIPTION
+    Prepares the MRTK\Extensions folder for UPM packaging.
+.PARAMETER PackageRoot
+    The root folder containing the examples package contents. If not specified, the current folder is presumed.
+#>
+param(
+    [string]$PackageRoot
+)
+
+if (-not $PackageRoot) {
+    throw "Unknown package root path. Please specify -PackageRoot when building."
+}
+
+# This hashtable contains mappings of the sample categories to the folder which contains
+# the code and assets.
+#
+# Note that capitalization below in the key itself is significant.
+#
+# These paths are PackageRoot relative.
+$exampleFolders = [ordered]@{
+    "HandPhysicsService" = "HandPhysicsService\Examples";
+}
+
+# Beginning of the upm packaging script main section
+# The overall structure of this script is:
+#
+# 1) Create the required Samples~ folder
+# 2) Recursively copy the contents of the example folders into Samples~
+# 3) Prepare the samples entry for the package.json file
+# 4) Replace the %samples% token with the prepared data
+# 5) Overwrite the package.json file
+
+# Ensure the required folder exists
+$samplesFolder = "$PackageRoot\Samples~"
+if (-not (Test-Path -Path $samplesFolder)) {
+    New-Item $samplesFolder -ItemType Directory | Out-Null
+}
+
+# Copy each example folder
+foreach ($entry in $exampleFolders.GetEnumerator()) {
+    $sampleGroupName = $entry.Name
+    $sampleFolder = $entry.Value
+    Write-Output "Copying $PackageRoot\$sampleGFolder to $samplesFolder\$sampleGroupName"
+    Copy-Item -Path "$PackageRoot\$sampleFolder" -Destination "$samplesFolder\$sampleGroupName" -Recurse -Force
+    Copy-Item -Path "$PackageRoot\$sampleFolder.meta"-Destination "$samplesFolder\$sampleGroupName.meta"
+}
+
+# Create the samples data for the package.json file
+$sampleCount = 0
+$samples = "`"samples`": ["
+foreach ($entry in $exampleFolders.GetEnumerator()) {

```

Comment:
Out of curiousity, why did this code opt to operate on json files by manually doing file edits, rather than just using powershell's JSON functionality?

i.e. we could read it in, add the samples node, write it out, and it would I assume also then auto-format and do other stuff.

Reply:
It's just odd to see manual file manipulation of types like JSON/XML that have decently robust libraries in modern scripting languages.

I haven't tried this myself in powershell so maybe powershell's things actually are pretty crappy for this (I'd believe that)

Reply:
tbh, i wasnt aware of the ps json support. will investigate

Reply:
we discussed this "in person". given that the manifests have been renamed as templates, and this code works, we will leave it as is.